railib/api.py

```diff
@@ -325,6 +326,38 @@ def _parse_arrow_results(files: List[TransactionAsyncFile]):
             results.append({"relationId": file.name, "table": table})
     return results
 
+# polling with specified overhead
+# delay is the overhead % of the time the transaction has been running so far
+
+
+def poll_with_specified_overhead(
+    f,
+    overhead_rate: float = 0.1,
+    start_time: datetime = datetime.now(timezone.utc),
+    timeout: int = None,
+    max_retries: int = None
+):
+    retries = 0
+    max_time = time.time() + timeout if timeout else None
+
+    while True:
+        if max_retries is not None and retries >= max_retries:
+            raise Exception(f'max retries {max_retries} exhausted')
+
+        if max_time is not None and time.time() >= max_time:
+            raise Exception(f'timed out after {timeout} seconds')
+
+        if f():
+            break
+
+        retries += 1
+        duration = (datetime.now(timezone.utc) - start_time).total_seconds() * overhead_rate
```

Comment:
Why do we use `time.time()` above and `datetime.now()` here?

Reply:
we use `datetime` to convert transaction `created_on` timestamp to `utc datetime` (the comparison is based on utc timezone) which will require extra processing if using `time.time()`
we simply use `time.time()` to check timeout, we can also use `datetime`